### PR TITLE
fix: optimize some APIs

### PR DESCRIPTION
**Pull request by Beacontownfc**
`tf.nn.moments` : There may be a problem where the gradient is `null` during the training,  it seems that removing the 681st line of code from the `[TensorFlowNET.Core/Operations/array_ops.cs]` can solve this issue.

`tf.nn.batch_normalization`: Use `nn_impl.batch_normalization` to reuse code.

### CODE REVIEW

File: src/TensorFlowNET.Core/Operations/array_ops.cs [hunk 1]
```diff
@@ -678,7 +678,6 @@ public static Tensor stop_gradient(Tensor input, string name = null)
             var tape = tf.GradientTape().stop_recording();
             var result = gen_array_ops.stop_gradient(input, name);
             tape.StartRecord();
-            tf.GradientTape().PushTape(tape);
```
Comment (OliBomby) [inline src/TensorFlowNET.Core/Operations/array_ops.cs:681]: this removal looks highly suspicious. Does anyone even know why this statement is here?

Comment (Oceania2018) [reply src/TensorFlowNET.Core/Operations/array_ops.cs:681]: Everything looks fine when I run though all the [examples](https://github.com/SciSharp/SciSharp-Stack-Examples).

Comment (OliBomby) [reply src/TensorFlowNET.Core/Operations/array_ops.cs:681]: none of the examples test this specific method though

Comment (Beacontownfc) [reply src/TensorFlowNET.Core/Operations/array_ops.cs:681]: Not deleting this line of code will result in `NULL` gradient when using the `stop_gradient` API.

Comment (Beacontownfc) [reply src/TensorFlowNET.Core/Operations/array_ops.cs:681]: [LayerNorm](https://github.com/SciSharp/TensorFlow.NET/blob/master/src/TensorFlowNET.Keras/Layers/Normalization/LayerNormalization.cs) uses `tf.nn.moment` and `tf.nn.moment` uses the `stop_gradient` API. If this line of code is not deleted, an error will be reported. Deleting this line of code will enable normal training.
